### PR TITLE
Feature/data model

### PR DESCRIPTION
**Pull request by cliffxuan**
1. initial data model
2. minimal api client for float
   entity-diagram

### CODE REVIEW

File: dashboard/apps/prototype/models.py [hunk 1]
```diff
@@ -1,3 +1,87 @@
...
+
+    class Meta:
+        unique_together = ('name', 'is_contractor')
+
+
+class Rate(models.Model):
+    amount = models.DecimalField(max_digits=5, decimal_places=2)
+    person = models.ForeignKey('Person', related_name='rates')
+    start_date = models.DateTimeField()
+    end_date = models.DateTimeField()
```
Comment (s-block) [inline dashboard/apps/prototype/models.py]: maybe set null=True on end_date as an end_date won't be know

you could even get rid of end_date all together as the end date will be the start date of the later entry

File: dashboard/apps/prototype/models.py [hunk 2]
```diff
@@ -1,3 +1,87 @@
...
+        else:
+            return '"{}" @ "{}"/day from "{}"'.format(
+                self.person, self.amount, self.start_date)
+
+
+class PersonRole(models.Model):
+    person = models.ForeignKey('Person')
+    role = models.ForeignKey('Role')
+    start_date = models.DateTimeField()
+    end_date = models.DateTimeField()
```
Comment (s-block) [inline dashboard/apps/prototype/models.py]: end_date for the role won't be known when the PersonRole is created so would have to be null=True

Or remove end_date and use the start_date of the later record as the end_date

Comment (cliffxuan) [inline dashboard/apps/prototype/models.py]: good point. will remove `end_date`

File: dashboard/apps/prototype/models.py (hunk 1 above)
Comment (cliffxuan) [inline dashboard/apps/prototype/models.py]: good point. going to remove the `end_date`

**Comment by cliffxuan** [comment]
updated diagram. added a few `float_id`s and removed some `end_date`s.

entity-diagram

**Comment by cliffxuan** [comment]
latest
entity-diagram

**Pull request merged**